### DISCUSSION TITLE
export names as they are

### DISCUSSION THREAD

**Discussion by womeier** [wasm-bindgen/wasm-bindgen#3911] [Q&A] [answered]
### Background
I'm compiling multiple programs to Wasm and want to run them with the same Node.js-based runner script, my setup is similar to the following. The generated Wasm binary should import a print function `write_string` so it can write strings to stdout:

```
#[wasm_bindgen]
extern "C" {
  fn write_string(s: &str);
}

#[wasm_bindgen]
pub fn main() {
  write_string(&format!("{:?}", demo1()))
}
```

This generates a Wasm module expecting a function import under the name `__wbg_writestring_6842f18ee0be6fba`, where the hash-value changes depending on the concrete source program.

### Question
Is there an option so `wasm-bindgen` doesn't generate this hash-value in the first place?

I currently use a script to get rid of that, which works but is a bit unpleasant,
I couldn't find a flag for that in the [documentation](https://rustwasm.github.io/wasm-bindgen/).

Thanks!

**Comment by daxpedda** [comment, marked as answer]
I'm not aware of `wasm-bindgen` supporting this.
To circumvent `wasm-bindgen` you could just use Rust directly:
```rust
extern "C" {
    fn write_string(s: &str);
}
```
Obviously this has a lot of problems.

Why exactly do you need the imported name not to change?
The reason why `wasm-bindgen` generates a shim is  because e.g. `&str` isn't FFI safe and isn't compatible with JS, so this function can't really be exposed through Wasm in the first place.

**Reply by womeier** [reply]
Thanks for the response!

I'm not distributing the generated module to be used in another Node.js application,
we're just doing a performance comparison with another tool we built, we compare:
Coq ---[coq-rust-extraction]---> Rust --[wasm-pack,wasm-bindgen]---> Wasm
Coq ---[certicoq-wasm] ---> Wasm

We use this run-script: [Link](https://github.com/womeier/certicoqwasm-testing/blob/master/evaluation/rust_to_wasm/run-node.js). It is just more convenient for benchmarking to have one run-script instead of many.
But I have a working setup, so all good. I was just wondering if it was possible to simplify it.
